File: src/main/java/com/oz/office_tastezip/common/response/ResponseFail.java
package com.oz.office_tastezip.common.response;

import org.springframework.http.HttpStatus;
import org.springframework.util.StringUtils;

public class ResponseFail<T> extends Response<T> {

    private ResponseCode responseCode;
    private String errorMessage;

    public ResponseFail() {
    }

    public ResponseFail(ResponseCode responseCode) {
        this.responseCode = responseCode;
    }

    public ResponseFail(ResponseCode responseCode, String errorMessage) {
        this.responseCode = responseCode;
        this.errorMessage = errorMessage;
    }

    @Override
    protected String resultCode() {
        return responseCode.getCode();
    }

    @Override
    protected String resultMessage() {
        return StringUtils.hasText(errorMessage) ? errorMessage : responseCode.getMessage();
    }

    @Override
    protected HttpStatus resultHttpStatus() {
        return responseCode.getHttpStatus();
    }

}

File: src/main/kotlin/com/oz/office_tastezip/global/response/ResponseFail.kt
package com.oz.office_tastezip.global.response

import org.springframework.http.HttpStatus
import org.springframework.util.StringUtils

class ResponseFail<T>(
    private val responseCode: ResponseCode = ResponseCode.FAIL,
    private val errorMessage: String? = null
) : Response<T>() {

    override fun resultCode(): String = responseCode.code

    override fun resultMessage(): String =
        if (StringUtils.hasText(errorMessage)) errorMessage!! else responseCode.message

    override fun resultHttpStatus(): HttpStatus = responseCode.httpStatus
}
